### PR TITLE
combined tick and draw loops

### PR DESCRIPTION
**Pull request by Tombez**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -93,29 +94,30 @@ window.addEventListener('load', () => {
   toggleBtn.addEventListener('click', () => {
     if(canvas.hasAttribute('data-paused')) {
```
Comment (kamoroso94) [inline main.js]: We can probably get rid of the `[data-paused]` since there's a boolean flag now.

File: main.js [hunk 2]
```diff
@@ -93,29 +94,30 @@ window.addEventListener('load', () => {
...
-  tickId = setTimeout(tick, 1000 / TPS);
+  lastTick = performance.now();
+  requestAnimationFrame(loop);
+
+  function loop(now) {
+    const dt = (now - lastTick) / 1000;
+    draw();
+    if (!paused) tick(dt);
+    lastTick = now;
+  }
```
Comment (kamoroso94) [inline main.js:113]: Are you forgetting a call to `requestAnimationFrame` to continue the loop?

**Review by kamoroso94 — CHANGES_REQUESTED** [review]
Check out my comments about what may need to be changed before merging.

**Comment by Tombez** [comment]
Looks like I accidentally removed it.

File: main.js (hunk 2 above)
Comment (kamoroso94) [reply main.js:113]: Nice work :)

File: main.js (hunk 1 above)
Comment (kamoroso94) [reply main.js]: Awesome!

**Comment by kamoroso94** [comment]
Great changes!

**Pull request merged**